docs/DeveloperGuide.md

```diff
@@ -4,11 +4,67 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
+## Class Structure
```

Comment:
In general, good job for clear and precise categorization of the contents inside the developer guide!